test/cpp/api/CMakeLists.txt

```diff
@@ -94,3 +102,12 @@ endif()
 add_executable(parallel_benchmark ${TORCH_API_TEST_DIR}/parallel_benchmark.cpp)
 target_include_directories(parallel_benchmark PRIVATE ${ATen_CPU_INCLUDE})
 target_link_libraries(parallel_benchmark PRIVATE torch)
+
+if(USE_DEPLOY)
```

Comment:
so, you shouldn't have to do this since it is already being called by our CI in this script:

https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L456

note, as I see this I am wondering why it was called in the CI script instead of cmake in the first place.  it might be better to move it here.  But since there might have been a reason for doing it that way and since it should already be available for you, i'd suggest doing that change in a separate PR if you want to do it.

Reply:
oh, and if we do move the generate step into cmake, it should be done just once, in the torchdeploy CMakeLists rather than here.  We should still be able to get access to it somehow from here.

Reply:
I noticed the CI script as well. Maybe that step is done in the CI script instead of the CMake rules is due to the CI failures that I'm having here.

The reason why I did it in the CMake is that if I'm running the test like: `./build/bin/test_api --gtest_filter=IMethodTest.*`, it doesn't seem to be able to trigger the CI script. Maybe I ran the test in a wrong way.

Reply:
well the way it is set up now, you would have to run the generate step manually, it doesn't make sense to run the whole CI script manually.  so i favor moving it to CMakeLists.

Reply:
Then let me spilt the patch into two:

1. Just includes the test without invoking the generate_examples.py.
2. Tries to move the generate_examples.py to CMakeLists.

Reply:
yea, when you move it please try adding in one place (near test_deploy)